200.py

```diff
@@ -0,0 +1,23 @@
+class Solution:
+    def numIslands(self, grid: List[List[str]]) -> int:
+        count = 0
+
+        for i in range(len(grid)):
+            for j in range(len(grid[0])):
+                if grid[i][j] == '1':
+                    count += 1
+
+                self.dfs(grid, i, j)
+
+        return count
+
+    def dfs(self, grid: List[List[str]], row: int, col: int) -> None:
+        if row < 0 or row >= len(grid) or col < 0 or col >= len(grid[0]) or grid[row][col] != '1':
```

Comment:
This line is too long. I would break them into two categories:

1. Index out of bound (i.e., position not valid in `grid`)
2. Value does not meet our requirement (i.e., `grid[row][col] != '1'`)